src/MongoDB.Driver/FieldValueSerializerHelper.cs

```diff
@@ -28,16 +28,11 @@ internal static class FieldValueSerializerHelper
     {
         public static IBsonSerializer GetSerializerForValueType(IBsonSerializer fieldSerializer, IBsonSerializerRegistry serializerRegistry, Type valueType)
         {
-            return GetSerializerForValueType(fieldSerializer, serializerRegistry, valueType, 0);
+            return GetSerializerForValueType(fieldSerializer, serializerRegistry, valueType, allowScalarValueForArrayField: false);
```

Comment:
The "recursionLevel" used to be a subtle way of indicating (sort of) whether scalar values where allowed for array fields. 0 meant no and 1 meant yes.

This new parameter is more explicit.